server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -831,7 +811,7 @@ public static void main(String[] args) throws Exception {
   @Override
   public List<ActiveCompaction> getActiveCompactions(TInfo tinfo, TCredentials credentials)
       throws ThriftSecurityException, TException {
-    if (!security.canPerformSystemActions(credentials)) {
+    if (!security.get().canPerformSystemActions(credentials)) {
```

Comment:
Would it be possible to do the following and drop the `security` instance variable?

```suggestion
    if (!getContext().getSecurityOperation().canPerformSystemActions(credentials)) {
```

Reply:
It looks like the answer is yes. I didn't think about that, because it seemed like it was being done this way for a test or something (so you can override the security object for testing), but I don't actually see anything like that being done.